src/components/commons/Modal.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState, useEffect, Dispatch, SetStateAction, useRef, forwardRef } from "react";
+import { Swiper, SwiperSlide } from "swiper/react";
+import "swiper/css";
+import { Swiper as SwiperType } from "swiper"; // Swiper 타입 임포트
+import { ModalWrapper, ModalBox, ModalCloseButton, ModalItemWrapper, ModalVideoWrapper, ToonWrapper, Pagination, ModalControl } from "@styles/commons/modal.style";
+interface ModalProps {
+    type: string;
+    itemId: number;
+    modalOpen: boolean;
+    setModalOpen: Dispatch<SetStateAction<boolean>>;
+}
+
+function Modal({ type, itemId, modalOpen, setModalOpen }: ModalProps) {
```

Comment:
```suggestion
import type { FC } from "react";

// 선언식
function Modal({ type, itemId, modalOpen, setModalOpen }: ModalProps): FC {
...

// 표현식
const Modal: FC<ModalProps> = ({ type, itemId, modalOpen, setModalOpen }) => {
  ...
```

타입스크립트를 사용할 때는 선언식보다 표현식이 가독성이 더 좋다고 생각합니다.